modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Keywords.js

```diff
@@ -38,122 +39,197 @@ export default function Keywords({currentPage, languageTag}) {
 
 	const [{publishedToday}] = useContext(StoreContext);
 
+	const keywordsData = useMemo(() => {
+		const dataKeys = new Set();
+
+		const countries = currentPage.data.countryKeywords.reduce(
+			(acc, {countryCode, countryName}) => {
+				if (dataKeys.has(countryCode)) {
+					return acc;
+				}
+
+				dataKeys.add(countryCode);
+
+				return acc.concat({countryCode, countryName});
+			},
+			[]
+		);
+
+		return {countries};
+	}, [currentPage.data.countryKeywords]);
+
+	const {countries} = keywordsData;
+
+	const [currentCountry, setCurrentCountry] = useState(
+		countries[0].countryCode
+	);
+
+	const countryKeywords = useMemo(() => {
+		const country =
+			currentCountry &&
+			currentPage.data.countryKeywords.find((c) => {
```

Comment:
We usually try to avoid single character (or abbreviations) variable names with the exception of counter iterators like the `i` or `j` in a `for` loop.

Reply:
If countries where to be a map, this .find() call could be replaced by a faster object key lookup:

```js
const country = countries[currentCountry];
...
```

Reply:
Ok, I'll change the single characters from the code. In the other hand, `countries` is intended to use it in the `<ClaySelect>` (format: [{countryCode: 'us', countryName: 'United States']) and in this point of the code I'm finding the keywords for that country (which are inside `currentPage.data.countryKeywords` as an array of objects cointaing the country information and the keywords). Also, I'm going to rename the const name to `const countryKeywords = ...`.

Reply:
```js
const keywords = useMemo(() => {
	const countryKeywords =
		currentCountry &&
		currentPage.data.countryKeywords.find((country) => {
			return country.countryCode === currentCountry;
		});

	return countryKeywords?.keywords ?? [];
}, [currentPage.data.countryKeywords, currentCountry]);
```

Reply:
Right! Got it!